### ISSUE TITLE
Orientation prop buggy

### ISSUE THREAD

**Issue by StrikoMirko** [airbnb/react-dates#242]
Hi guys,

I'm experiencing some buggy behaviour with the orientation prop.
What I needed logic that will re-render the DateRangePicker if the window gets resized, this is for mobile rotation. So if a user is on landscape mobile and rotates it to normal then DateRangePicker should re-render and display itself in 'vertical' orientation and vice versa.
My threshold is 600 width. 
DateRangePicker re-renders correctly when i resize the window under 600 width and the orientation is horizontal but when I start to expand the window over 600 and DateRangePicker should now re-render from vertical to horizontal then the component breaks, actually it renders just a border with the rest of it invisible.

Basically the component breaks if you dynamically change the orientation prop from vertical to horizontal. I do this as a listener on window.resize.

I would appreciate if anyone could check this.

Thank you.

**Comment by jzhang300** [comment]
I'm trying to do what you're doing basically and am experiencing same issue.  When the orientation changes on window resize, the absolute positioning of the datepicker component does not properly update:

![screen shot 2017-01-09 at 2 46 31 pm](https://cloud.githubusercontent.com/assets/1449523/21782820/725c9fe4-d67a-11e6-8840-345e34bbef0c.png)

**Comment by jzhang300** [comment]
Also when changing orientation from vertical to horizontal when app is running, the background of the date component disappears.

![screen shot 2017-01-09 at 3 50 41 pm](https://cloud.githubusercontent.com/assets/1449523/21784984/66748170-d683-11e6-9b9d-f5b7feee6cca.png)

A temporary fix is to give `.transition-container` `min-height: 313px;`:

![screen shot 2017-01-09 at 3 53 59 pm](https://cloud.githubusercontent.com/assets/1449523/21785129/df471676-d683-11e6-8530-4a0c9dbb3e67.png)

**Comment by majapw** [comment]
I will look into this. I think that switching from landscape to portrait on a phone is a compelling argument to support resizing properly. 

@jzhang300 @StrikoMirko Both of you are changing the orientation from vertical in portrait mode to horizontal in landscape mode, correct?

**Comment by StrikoMirko** [comment]
Yes correct. When changing from horizontal to vertical then the calendar is rendered normally but from vertical to horizontal breaks it.

**Comment by jzhang300** [comment]
Transitioning from both vertical to horizontal and horizontal to vertical have their own issues.  From horizontal to vertical, the positioning of calendar may not be correct.  From vertical to horizontal, the background of the calendar disappears because `.transition-container` has `0` height.

**Comment by piotrkosytorz** [comment]
+1

**Comment by ljharb** [comment]
@piotrkosytorz please do not comment anywhere on github with "+1 " - use an emoji reaction on the original post, to avoid creating clutter. I'll delete your comment and my reply to keep the thread clean.

**Comment by piotrkosytorz** [comment]
Is there is a plan to debug it, or has anyone come up with a fix?

**Comment by majapw** [comment]
I think https://github.com/airbnb/react-dates/pull/446 should address this issue but it's been stagnant for a while. I can take it over and try to do a release this week.
